src/Microsoft.Azure.ServiceBus/Amqp/AmqpMessageConverter.cs

```diff
@@ -362,6 +363,90 @@ public static AmqpMap GetRuleDescriptionMap(RuleDescription description)
             return ruleDescriptionMap;
         }
 
+        public static RuleDescription GetRuleDescription(AmqpRuleDescriptionCodec amqpDescription)
+        {
+            Filter filter = GetFilter(amqpDescription.Filter);
+            RuleAction action = GetRuleAction(amqpDescription.Action);
+
+            RuleDescription description = new RuleDescription(filter)
+            {
+                Action = action,
+                Name = amqpDescription.RuleName
+            };
+
+            return description;
+        }
+
+        public static Filter GetFilter(AmqpFilterCodec amqpFilter)
+        {
+            Filter filter;
+
+            if (amqpFilter.DescriptorCode == AmqpSqlFilterCodec.Code)
```

Comment:
+1

---
In reply to: [128698341](https://github.com/Azure/azure-service-bus-dotnet/pull/192#discussion_r128698341) [](ancestors = 128698341)